src/deque.jl

```diff
@@ -80,6 +80,16 @@ Create a deque of type `T`.
 """
 deque(::Type{T}) where {T} = Deque{T}()
 
+# constructor for iterator
+function Deque(iter)
+    n = length(iter)
```

Comment:
`n` isn't used here.  Did you mean to use it as a size parameter when constructing the `Deque`?  

Obviously, if you use `deque`, that isn't possible, but if you call the main constructor, you can.

That said, since iterators don't necessarily have a known length, I would suggest making it an optional parameter to the function.

```
function Deque(iter, blksize::Int=DEFAULT_DEQUEUE_BLOCKSIZE)
```
and then modify the deque construction below accordingly.